weasyl/emailer.py

```diff
@@ -25,6 +25,29 @@ def normalize_address(address):
     return "%s@%s" % (local, domain.lower())
 
 
+def check_email_blacklist(address):
+    """
+    Determines if a supplied email address is present in the 'emailblacklist' table.
+
+    Parameters:
+        address: The email address to split out the domain from.
+
+    Returns:
+        Boolean True if present on the blacklist, or False otherwise.
+    """
+    address = address.strip()
+
+    if not EMAIL_ADDRESS.match(address):
+        return None
+
+    local, domain = address.split("@", 1)
```

Comment:
Nobody actually has e-mail addresses like this (and I’m not sure that we allow them), but please use `rsplit` anyway as a matter of good practice. `"this@is"@valid`.